tests/components/tolosauna/test_config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Tests for the TOLO Sauna config flow."""
+from unittest.mock import Mock, patch
+
+import pytest
+
+from homeassistant.components.dhcp import IP_ADDRESS, MAC_ADDRESS
+from homeassistant.components.tolosauna.const import DOMAIN
+from homeassistant.config_entries import SOURCE_DHCP, SOURCE_USER
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+MOCK_DHCP_DATA = {IP_ADDRESS: "127.0.0.2", MAC_ADDRESS: "00:11:22:33:44:55"}
+
+
+@pytest.fixture(name="toloclient")
+def toloclient_fixture() -> Mock:
+    """Patch libraries."""
+    with patch(
+        "homeassistant.components.tolosauna.config_flow.ToloClient"
+    ) as toloclient:
+        yield toloclient
+
+
+async def test_user_no_input(hass: HomeAssistant):
```

Comment:
The config flow is being tested in parts. Please add a test that covers the full flow walkthrough.

Reply:
Which part is missing? For me, coverage report reports 100%:
```
$ pytest ./tests/components/tolosauna/test_config_flow.py --cov=homeassistant.components.tolosauna --cov-report term-missing -vv
[...]
---------- coverage: platform linux, python 3.8.12-final-0 -----------
Name                                                Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------
homeassistant/components/tolosauna/config_flow.py      45      0   100%
homeassistant/components/tolosauna/const.py             8      0   100%
---------------------------------------------------------------------------------
TOTAL                                                  53      0   100%

```

Reply:
Yes, my comment wasn't about the coverage. Coverage just means all lines are hit by tests, however, that doesn't mean the test is good or not (if that makes sense? Please don't be offended by it, that is not what I mean).

Here is an example of a config flow that is fully walked through:

<https://github.com/home-assistant/core/blob/593bc866f049997f0b4cf7c3ceee29cd1468d7c9/tests/components/rdw/test_config_flow.py#L13-L34>

So you now have a test that tests if a form shows up, a tests to see if an entry is created, and a test an error is shown. But no test that tests the combination of all these things start user flow -> integration setup.

Reply:
I'm not offended at all, on the contrary, I am happy to learn something - just did not know what is missing here. So, thx for your patience and hints! :) I think I got the point here, will try to improve it and come back to you.

Reply:
I pushed a new test covering the full user flow (https://github.com/home-assistant/core/pull/55619/commits/2b2c2d38abfcc0b231e7d6ccbe8ec7ca23fa85ea). Could you please check if that's what you meant? If yes, you could mark this as resolved :) Thanks again!